roles/custom/matrix-dendrite/tasks/setup_install.yml

```diff
@@ -72,6 +88,11 @@
     owner: "{{ matrix_user_username }}"
     group: "{{ matrix_user_groupname }}"
 
+- name: Ensure Dendrite Docker image is built
+  ansible.builtin.command:
+    cmd: "{{ devture_systemd_docker_base_host_command_docker }} build -t {{ matrix_dendrite_docker_image }} {{ matrix_dendrite_docker_src_files_path }}"
+  when: "matrix_dendrite_container_image_self_build | bool"
+
```

Comment:
You may wish to get inspired from the `matrix-synapse` role, where we also check if the image has been built before trying to fire the `docker build` command: https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/99f69822877a65d55b42a0755af34e124cea8e53/roles/custom/matrix-synapse/tasks/synapse/setup_install.yml#L21-L52

I'm not sure if this is strictly necessary though. If the files in `matrix_dendrite_docker_src_files_path` remain unchanged, it will likely not rebuild the image again anyway?

Reply:
@spantaleev thanks, good call. I looked into it a bit, and while it's true that Docker will figure out that it needn't rebuild the images again, ansible will still run the command and mark the task as "changed". So this system gives us a more accurate ansible response, as well as makes the overall script more efficient. See https://github.com/spantaleev/matrix-docker-ansible-deploy/pull/2674/commits/bc709217cfebc49b729cec14364c62d280288875